Min broker children (#1969)
GMLC-TDC/HELICS: docs/references/configuration_options_reference.md

String used to define the configuration of the broker if one is autogenerated. Such configuration typically includes things like broker IP addresses and port numbers. Again, if this is a co-simulation running on a single computer (and is the only HELICS co-simulation running on said computer) the default option is likely to be sufficient. The following options are available for this string:

> - `--federates=` - The minimum Number of federates expected to join a federation though the broker. Equivalent to `-f`
> - `--max_federates=` - The maximum Number of federates allowed to join through a broker.

Reviewer: ```suggestion
- `--federates=` - The minimum number of federates expected to join a federation through the broker. Equivalent to `-f`.
- `--max_federates=` - The maximum number of federates allowed to join through a broker.
```

- `--timeout=` milliseconds to wait for all the federates to connect to the broker (can also be entered as a time like '10s' or '45ms')
- `--network_timeout=` - Time to establish a socket connection in ms. Times can also be entered as strings such as "15s" or "75ms".
- `--error_timeout=` - Time in ms to wait after an error state is reached before terminating. Times can also be entered as strings such as "15s" or "75ms".
- `--query_timeout=` - Time in ms to wait for a query to complete. Times can also be entered as strings such as "15s" or "75ms".

> - `--children=` - The minimum number of child object the broker should expect before allowing entry to initializing state.

Reviewer: ```suggestion
- `--children=` - The minimum number of child objects the broker should expect before allowing entry to the initializing state.
```

> - `--children=` - The minimum number of child objects the broker should expect before allowing entry to the initializing state.

Reviewer 2: I would prefer just one term, probably `subbrokers`.  If you think there will be Intergrators or Modelers who need the more specific term of `children` I could live with that.

Author: I needed some term that applied to cores as well as brokers.  Subbrokers doesn't make sense on cores but it is the most obvious on brokers.

> - `--subbrokers=` - The minimum number of child object the broker should expect before allowing entry to initializing state. Same as `--children` but might be clear in some case with multilevel hierarchies.

Reviewer: ```suggestion
- `--subbrokers=` - The minimum number of child objects the broker should expect before allowing entry to the initializing state. Same as `--children` but might be clearer in some cases with multilevel hierarchies.
```

> - `--brokerkey=` - A broker key to use for connections

Reviewer 2: Can we add more help details on `brokerkey`?  Could we add "... to ensure federates are connecting with a specific broker."?
